src/etc/inc/captiveportal.inc

```diff
@@ -383,7 +384,8 @@ EOD;
 			captiveportal_send_server_accounting('on');
 			echo "done\n";
 
-			if (isset($cpcfg['preservedb'])) {
+			if (isset($cpcfg['preservedb']) || xmlrpc_sync_get_details($syncip, $syncport, $syncuser, $syncpass)) {
```

Comment:
This line is very long, move the last test down to the its own line (indented four spaces)